quarkchain/p2p/p2p_server.py

```diff
@@ -158,10 +158,17 @@ def _get_discv5_topic(self) -> bytes:
             await self._receive_handshake(reader, writer)
         except expected_exceptions as e:
             self.logger.debug("Could not complete handshake: %s", e)
+            self.logger.error("Could not complete handshake: %s", e)
+            reader.feed_eof()
+            writer.close()
         except OperationCancelled:
-            pass
+            self.logger.error("OperationCancelled")
+            reader.feed_eof()
+            writer.close()
         except Exception as e:
             self.logger.exception("Unexpected error handling handshake")
+            reader.feed_eof()
```

Comment:
can use `finally` block

Reply:
finally will close writer regardless of exception??

Reply:
yup

Reply:
that's not right, shall not close if no exception

Reply:
ah i see. another way to save those duplicate lines would be return at line 158 and close after the try-catch block